lib/av_client/crypto/pedersen_commitment.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  addPoints,
+  Bignum,
+  bignumFromHex,
+  bignumToHex,
+  Curve,
+  generateRandomBignum,
+  hashToBignum,
+  Point,
+  pointFromHex,
+  pointFromX,
+  pointToHex,
+} from "./util";
+
+class PedersenCommitment {
+  static computeGenerator(index) {
+    const baseGeneratorPrefix = () => pointToHex(new Point(Curve.G));
+    const secp256k1_curve_prime = new Bignum(Curve.field.modulus);
+
+    const target = [baseGeneratorPrefix(), index].join(',');
+    let x = hashToBignum(target).mod(secp256k1_curve_prime);
+
+    /*eslint no-constant-condition: ["error", { "checkLoops": false }]*/
+    while(true) {
+      try {
+        return pointFromX(x);
+      }
+      catch(err) {
+        // Skip to next iteration
+      }
+
+      x = x.add(new Bignum(1)).mod(secp256k1_curve_prime);
+    }
+  }
+
+  static verify(commitment: Point, messages, randomizer: Bignum) {
+    return commitment.equals(this.generate(messages, randomizer))
+  }
+
+  static generate(messages: Bignum[], randomizer: Bignum): Point {
+    let commitment = new Point(Curve.G).mult(randomizer);
+
+    messages.forEach((message, index) => {
+      const generator = PedersenCommitment.computeGenerator(index);
+      const term = generator.mult(message);
+
+      commitment = addPoints(commitment, term);
+    });
+
+    return commitment;
+  }
+}
+
+/**
+ * @description Generates an encryption commitment
+ * @param messages An array of hex strings
+ * @param options Optional options object. Allows caller to specify randomizer
+ * @returns Commitment point and randomizer, both as hex.
+ */
+const generatePedersenCommitment = (messages: string[], options?: { randomizer: string }) => {
+  const bnMessages: Bignum[] = messages.map(m => bignumFromHex(m));
+
+  const randomizer = options && options.randomizer ?
+    new Bignum(options.randomizer) : generateRandomBignum();
+
+  const commitment = PedersenCommitment.generate(bnMessages, randomizer);
+
+  return {
+    commitment: pointToHex(commitment),
+    randomizer: bignumToHex(randomizer),
```

Comment:
Is `randomizer` the right name? It is very generic and sounds more like a verb (indicating function or object) than a noun.

Reply:
I agree. The trade-off is that this is the name used in our (academic) documentation.

Reply:
My understanding is that this corresponds to the cryptographic concept referred to as "a/the randomizer".

Reply:
Ok